samtranslator/model/api/api_generator.py

```diff
@@ -276,7 +280,58 @@ def _construct_api_domain(self, rest_api):
                 basepath_mapping.BasePath = path
                 basepath_resource_list.extend([basepath_mapping])
 
-        return domain, basepath_resource_list
+        # Create the Route53 RecordSetGroup resource
+        record_set_group = None
+        if self.domain.get('Route53') is not None:
+            route53 = self.domain.get('Route53')
+            if route53.get('HostedZoneId') is None:
+                raise InvalidResourceException(self.logical_id,
+                                               "HostedZoneId is required to enable Route53 support on Custom Domains.")
+            logical_id = logical_id_generator.LogicalIdGenerator("", route53.get('HostedZoneId')).gen()
+            record_set_group = Route53RecordSetGroup('RecordSetGroup' + logical_id,
+                                                     attributes=self.passthrough_resource_attributes)
+            record_set_group.HostedZoneId = route53.get('HostedZoneId')
+            record_set_group.RecordSets = self._construct_record_sets_for_domain(self.domain)
+
+        return domain, basepath_resource_list, record_set_group
+
+    def _construct_record_sets_for_domain(self, domain):
+        recordset_list = []
+        recordset = {}
+        route53 = domain.get('Route53')
+
+        recordset['Name'] = domain.get('DomainName')
+        recordset['Type'] = 'A'
+        recordset['AliasTarget'] = self._construct_alias_target(self.domain)
+        recordset_list.extend([recordset])
```

Comment:
Why extend here? Are we going to make recordset into a list later? I don't see any existing values for recordset_list.

Reply:
Just to keep it consistent and future proof (in case we add more types of recordsets in the future)